File: src/BestTimeToBuyAndSellStock.java
public class BestTimeToBuyAndSellStock {
    public static void main(String[] args) {
        int[] prices = {7, 1, 5, 3, 6, 4};
        int i = maxProfit(prices);
        System.out.println(i);
    }

    public static int maxProfit(int[] prices) {
        int buy = prices[0];
        int maxProfit = 0;
        for (int i = 1; i < prices.length; i++) {
            int price = prices[i];
            if (buy > price) {
                buy = price;
            } else if (price - buy > maxProfit) {
                maxProfit = price - buy;
            }
        }
        return maxProfit;
    }
}

File: src/javascript/BestTimeToBuyAndSellStock.js
const maxProfit = (prices) => {
    let buy = prices[0];
    let maxProfit = 0;
    for (let i = 1; i < prices.length; i++) {
        let price = prices[i];
        if (buy > price) {
            buy = price;
        } else if (price - buy > maxProfit) {
            maxProfit = price - buy;
        }
    }
    return maxProfit;
};

let prices = [7, 1, 5, 3, 6, 4];

let result = maxProfit(prices);
console.log(result);